client/cli/src/lib.rs

```diff
@@ -243,47 +247,34 @@ pub fn init_logger(
 	pattern: &str,
 	tracing_receiver: sc_tracing::TracingReceiver,
 	profiling_targets: Option<String>,
+	disable_log_reloading: bool,
 ) -> std::result::Result<(), String> {
-	fn parse_directives(dirs: impl AsRef<str>) -> Vec<Directive> {
-		dirs.as_ref()
-			.split(',')
-			.filter_map(|s| s.parse().ok())
-			.collect()
-	}
-
 	if let Err(e) = tracing_log::LogTracer::init() {
 		return Err(format!(
 			"Registering Substrate logger failed: {:}!", e
 		))
 	}
 
-	let mut env_filter = tracing_subscriber::EnvFilter::default()
-		// Disable info logging by default for some modules.
-		.add_directive("ws=off".parse().expect("provided directive is valid"))
-		.add_directive("yamux=off".parse().expect("provided directive is valid"))
-		.add_directive("cranelift_codegen=off".parse().expect("provided directive is valid"))
-		// Set warn logging by default for some modules.
-		.add_directive("cranelift_wasm=warn".parse().expect("provided directive is valid"))
-		.add_directive("hyper=warn".parse().expect("provided directive is valid"))
-		// Enable info for others.
-		.add_directive(tracing_subscriber::filter::LevelFilter::INFO.into());
+	// Initialize filter - ensure to use `add_default_directives` for any defaults to persist
+	// after log filter reloading by RPC
+	let mut env_filter = EnvFilter::default();
+	// Enable info
+	env_filter = add_default_directives(env_filter,"info", true);
+	// Disable info logging by default for some modules.
+	env_filter = add_default_directives(env_filter, "ws=off,yamux=off,cranelift_codegen=off", true);
+	// Set warn logging by default for some modules.
+	env_filter = add_default_directives(env_filter, "cranelift_wasm=warn,hyper=warn", true);
 
 	if let Ok(lvl) = std::env::var("RUST_LOG") {
 		if lvl != "" {
-			// We're not sure if log or tracing is available at this moment, so silently ignore the
-			// parse error.
-			for directive in parse_directives(lvl) {
-				env_filter = env_filter.add_directive(directive);
-			}
+			env_filter = add_default_directives(env_filter, &lvl, true);
```

Comment:
Here we would not panic!

Here we would just propagate the error!

Reply:
Should be resolved, see above.